projects/igniteui-angular/src/lib/core/utils.ts

```diff
@@ -98,10 +98,43 @@ export function isObject(value: any): boolean {
  * @returns true if provided variable is Date
  * @hidden
  */
-export function isDate(value: any) {
+export function isDate(value: any): boolean {
     return Object.prototype.toString.call(value) === '[object Date]';
 }
 
+/**
+ * Parse provided input to Date.
+ * @param value input to parse
+ * @returns Date if parse succeed or null
+ * @hidden
+ */
+export function parseDate(value: any): Date | null {
```

Comment:
I don't think you need to perform the type checking for number. Just reduce this to returning `new Date(value)` if the value is not instance of Date.

Reply:
I kept the check for `value`, otherwise for `null` values of `value`, the method would return `Invalid Date` or `01 Jan 1970`